src/app/editor/editor.component.ts

```diff
@@ -0,0 +1,162 @@
+  editor: any;
+
+  wasError: boolean = false;
+  msgError: string;
+
+  formats: Array<Array<string>> = [
+    ['bold', 'italic', 'underline', 'strikethrough'],
+    // ['size', 'color_text', 'color_fill'],
+    ['align_left', 'align_center', 'align_right'],
+    ['list_bulleted', 'list_numbered']
+  ];
+
+  annotations: Array<any>;
+
+  annotationMap = {
+    'bold': 'style/fontWeight=bold',
+    'italic': 'style/fontStyle=italic',
+    'underline': 'style/textDecoration=underline',
+    'strikethrough': 'style/textDecoration=line-through',
+    'align_left': 'paragraph/textAlign=left',
+    'align_center': 'paragraph/textAlign=center',
+    'align_right': 'paragraph/textAlign=right',
+    'list_bulleted': 'paragraph/listStyleType=unordered',
+    'list_numbered': 'paragraph/listStyleType=decimal'
+  };
+
+  buttons: Map<string, boolean> = new Map<string, boolean>();
+
+  constructor(
+    private swellrt: SwellRTService,
+    private route: ActivatedRoute
+    ) {
+      this.disableAllButtons();
+  }
+
+  get editorElem() {
+    return (<HTMLElement>document.querySelector('#editor-container > div'));
+  }
+
+  get title() {
+    return this._title && this._title.getValue();
+  }
+
+  set title(value) {
+    if (this._title) {
+      this._title.setValue(value);
+    }
+  }
+
+  updateAllButtons() {
+    for (let formatGroup of this.formats) {
+      for (let format of formatGroup) {
+        let [key, val] = this.annotationMap[format].split('=');
+        this.buttons[format] = this.annotations && (this.annotations[key] === val);
+      }
+    }
+  }
+
+  disableAllButtons() {
+    for (let formatGroup of this.formats) {
+      for (let format of formatGroup) {
+        this.buttons[format] = false;
+      }
+    }
+  }
+
+  ngOnInit() {
+
+    let widgets = {
+
+      'img-link' : {
+
+        onInit: function(parentElement, state) {
+          parentElement.innerHTML = `<img src="${state}">`;
+        },
+
+        onChangeState: function(parentElement, before, state) {
+          parentElement.innerHTML = `<img src="${state}">`;
+        }
+      }
+
+    };
+
+    let annotations =  {};
+
+    this.editor = this.swellrt.editor('editor-container', widgets, annotations);
+
+    this.swellrt.getUser().then(user => {
+
+      let id = this.route.snapshot.params['id'];
+      this.swellrt.open(id).then(cObject => {
+
+        cObject.addParticipant(this.swellrt.domain);
+
+        // Initialize the doc
+        if (!cObject.root.get('doc')) {
+          cObject.root.put('doc', cObject.createText(''));
+        }
+
+        // Initialize the doc's title
+        if (!cObject.root.get('doc-title')) {
+          cObject.root.put('doc-title', cObject.createString('New document'));
+        }
+
+        // Open the doc in the editor
+        this._title = cObject.root.get('doc-title');
+        this.editor.edit(cObject.root.get('doc'));
+
+        this.editor.onSelectionChanged((annotations) => {
+          this.annotations = annotations;
```

Comment:
To adapt the code to recent SwellRT API changes, you should change these two lines by:
```js
this.editor.onSelectionChanged((range.annotations) => {
  this.annotations = range.annotations;
```